src/backend/src/yaksur_pup.c

```diff
@@ -46,6 +46,13 @@ static int ipup(const void *inbuf, void *outbuf, uintptr_t count, yaksi_type_s *
                 request->backend.outattr.type == YAKSUR_PTR_TYPE__GPU) {
                 reqpriv->gpudriver_id = id;
                 break;
+            } else if (request->backend.inattr.type == YAKSUR_PTR_TYPE__MANAGED &&
+                       request->backend.outattr.type == YAKSUR_PTR_TYPE__MANAGED &&
+                       request->backend.inattr.device != -1 &&
+                       request->backend.outattr.device != -1 &&
+                       request->backend.inattr.device != request->backend.outattr.device) {
+                reqpriv->gpudriver_id = id;
+                break;
```

Comment:
I thought the intent of this PR was to handle cases where one buffer was on the GPU and the other was managed but could not migrate to the other GPU.  In such cases, since at least one of the buffers is on the GPU, the first "if" already handles this.

Reply:
ok, you are right. This case is not needed.

Reply:
There is a bunch of code in this PR that is following the same incorrect logic.  This PR needs several fixes.